### PR TITLE
Budget App: Final Review

### PR DESCRIPTION
**Pull request by smunozmo**
# Documentation

All the documentation is on the README file.

## Updates:

- Used ruby gems as software packages system.
- Used controllers to handle requests and render empty views.
- Used params from browser request in a safe way.
- Used preprocessed html file with embedded Ruby code.
- Used layouts and templates for shared content.
- Wrote SQL queries with ActiveRecord.
- Set up associations between models.
- Built a webapp that requires the user to log in.
- Used devise gem for authentication.
- Used integrity tests with RSPEC

## Important:

Use this credentials to see the test data:

email: user1@mail.com
password: 111111

## Screenshot

screenshot

## Live Demo

https://floating-thicket-39409.herokuapp.com/

## Video Demo

https://www.loom.com/share/a42a3acee19842c197a1e85ec4fe745f

## Notes:

✅  - Used  Ruby on Rails.
✅  - Used  PostgreSQL.
✅  - Used  Rubocop.
✅  - No linters errors.

### CODE REVIEW

File: app/views/groups/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+              <div class="col-3 d-flex align-items-center">
+                <div class="card-body">
+                  <p class="card-text"><small class="fs-5 p-3">$<%= GroupEntity.where("group_id = #{group.id}").joins(:entity).sum("amount") %></small></p>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+    <% end %>
+  <% end %>
```
Comment (Emmyn5600) [inline app/views/groups/show.html.erb]: - [x]  We don't need to add any logic inside the .erb file, we use the helper folders to create methods with logic and called inside the .erb file. You can refer to this comment to modify the files that have logic inside the .erb file.

Read [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make) for more information

File: app/views/entities/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                <div class="card-body">
+                  <p class="card-text"><small class="fs-5 p-3">$<%= Entity.find(entity.entity_id).amount %></small></p> 
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+    <% end %>
+  <% end %>
+</div>
```
Comment (Emmyn5600) [inline app/views/entities/show.html.erb]: - [x]  We don't need to add any logic inside the .erb file, we use the helper folders to create methods with logic and called inside the .erb file. You can refer to this comment to modify the files that have logic inside the .erb file.

Read [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make) for more information

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi, @smunozmo 

Good job so far! 👏🏼 

- Your project is running successfully.
- You have completed almost all requirements.
- Database is well structured.
- No linter errors
- You have a descriptive Readme file
- All tests are passing successfully.

Great work on making the project requirements:clap:
You've done well implementing the project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:

### Changes requested.

- [ ] Check the comment under the review.

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/helpers/entities_helper.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module EntitiesHelper
+  def entities_total(group)
+    GroupEntity.where("group_id = #{group.id}").joins(:entity).sum('amount')
+  end
```
Comment (sinansevgi) [inline app/helpers/entities_helper.rb:4]: [Optional]
- [x] You should avoid making SQL queries outside of models. It makes it harder to test and debug your app.

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @smunozmo, 

Great work on making the changes requested by a previous reviewer 👏 
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by smunozmo** [comment]
Thank you! @sinansevgi @Emmyn5600

**Pull request merged**